docs/storage-management.asciidoc

```diff
@@ -3,67 +3,159 @@
+For instance, a service with a popular tech stack (web framework, database, caching library, etc.) is more likely to generate bigger transactions.
+
+In addition, all agents support manual instrumentation.
+How little or much you use these APIs will also impact what a typical transaction looks like.
+
+Due to the high variability of transactions, it's difficult to classify a transaction as typical.
+Regardless, here's a speculative reference: 
+
+[options="header"]
+|=======================================================================
+|Transaction size |Number of Spans |Number of Frames
+|_Small_ |5-10 |5-10
+|_Medium_ |15-20 |15-20
+|_Large_ |30-40 |30-40
+|=======================================================================
+
+There will always be transaction outliers with hundreds of spans or frames, but those are very rare.
+Most customers should have _small_ transactions.
+
+[float]
+[[typical-storage]]
+=== Typical storage
+
+Consider the following typical storage reference.
+These numbers do not account for Elasticsearch compression.
+
+* 1 unsampled transaction is **~1 Kb**
+* 1 span with 10 frames is **~4 Kb**
+* 1 span with 50 frames is **~20 Kb**
+* 1 transaction with 10 spans, each with 10 frames is **~50 Kb**
+* 1 transaction with 25 spans, each with 25 spans is **250-300 Kb**
+* 100 transactions with 10 spans, each with 10 frames, sampled at 90% is **590 Kb**
+
+APM data compresses quite well, so the storage cost in Elasticsearch will be considerably less:
+
+* Indexing 100 unsampled transactions per second for 1 hour results in 360,000 documents. These documents use around **50 Mb** of disk space.   
+* Indexing 10 transactions per second for 1 hour, each transaction with 10 spans, each span with 10 frames, results in 396,000 documents. These documents use around **200 Mb** of disk space. 
+* Indexing 25 transactions per second for 1 hour, each transaction with 25 spans, each span with 25 frames, results in 2,340,000 documents. These documents use around **1.2 Gb** of disk space.
+
+NOTE: These examples were indexing the same data over and over with minimal variation. Because of that, the compression ratios of 80-90% are somewhat optimistic.
+
+[[processing-performance]]
+== Processing and performance
+
+APM Server performance depends on a number of factors: memory and CPU available,
+network latency, transaction sizes, workload patterns,
+agent and server settings, versions, and protocol.
+
+Let's look at a simple example that makes the following assumptions:
+
+* The load is generated in the same region as where APM Server and Elasticsearch are deployed.
+* We're using the default settings in cloud.
+* A small number of agents are reporting.
+
+This leaves us with relevant variables like payload and instance sizes.
+See the table below for approximations.
+As a reminder, events are
+{apm-overview-ref-v}/transactions.html[transactions] and
+{apm-overview-ref-v}/transaction-spans.html[spans].
+
+[options="header"]
```

Comment:
is great to have this here! we just need to remember to review/update each minor version :)